### ISSUE TITLE
Missing commentaries?

### ISSUE THREAD

**Issue by martinchristov** [fawazahmed0/quran-api#40]
I was hoping to use your good work for a mushaf app I am working on, but it's missing the commentaries of the translators. I consider these to be really important and want to feature them in my app. 
Could we work towards adding them?

**Comment by fawazahmed0** [comment]
While making this API, I avoided commentaries and I had many reason , for example people will avoid reading the quran altogether seeing huge size of commentary.
Other reason is because, the author may not give proper commentary. (For example there are authors of quran commentaries who have claimed themselves to be prophet and many other such things, it will be very tough to filter these type of commentaries)

I still don't intend to add commentaries due to above reasons

**Comment by martinchristov** [comment]
I see. Thank you for your response brother.
For my part I consider commentaries to be very important to be given as an optional additional text alongside ayat translations, as they have been an integral part of these translations. If you consider some of these authors not credible, then perhaps these authors should not have been included in the first place.

**Comment by fawazahmed0** [comment]
I don't include usually translations of [very controversial authors](https://github.com/fawazahmed0/quran-api/issues/35).

As we know, the muslim community is divided into different sects, each author who belongs to different sect, will try to interpret the verse in their commentary according to their beliefs and opinions, it will create a biased version of Quran.

Also it will be unjust if I cherry pick commentaries, which conform to my beliefs and opinions.

**Comment by martinchristov** [comment]
Interesting. I never even heard of Rashad Khalifa before. But nevermind.
I think it is good to filter out very controvertial translators.

However, if you do include the work of a scholar-translator, be he sunni or shia, I think that the whole book should be included - along with the contextual commentaries per surah and ayah.
For instance - in Bulgarian (my language) there is only one translation and it includes commentary which is of critical importance for a beginner-intermediate student of the Quran.
As for English - I would like to be able to compare translations AND their respective commentaries of scholars which I already trust.
Let the people decide which scholars they wish to trust. And whether they wish to read commentary or not.
If I were you I would include both Sunni and Shia scholars, while filtering out any really weird stuff.

In any case I want to feature that in my digital mushaf app, since I want to have this feature for myself.
If you at least have any advice/guidance how to get the translation + the commentary in some sort of digitised format, please send my way.

Peace and blessings

**Comment by fawazahmed0** [comment]
Do you have link to bulgarian translation and commentary?

Even though I will not include the commentary in this API, but I will convert it into JSON for you (if it's possible), so that you can use it in your App.

**Comment by martinchristov** [comment]
That is very kind of you.
Conversion to JSON I could do myself if I had some sort of digitised source file or database. Do you know where I can look for such, that might include the entire text of the authors, including commentary?

**Comment by husayt** [comment]
I would agree on including commentaries of translator. The reason is that Quran is Quran only in original arabic divine text, and every translation is mere interpretation of meanings by given translator. All of it including comment is a work of that translator and to give justice to work and for complete context it all should be included. So if translation is included all of it should be included, they are part of translation.  Of course, if commentaries contain anything dubious, then whole translation is dubious as well.

Also, thank you for avoiding unreliable or controversial translations.
